File: android/app/src/main/java/com/rbdarts/feature/stats/ProjectionService.kt
package com.rbdarts.feature.stats

import com.rbdarts.core.domain.ScoringRules

data class ProjectionResult(
    val label: String,
    val projectedFinalScore: Double
)

class ProjectionService {
    fun baseline(currentScore: Int, inningsPlayed: Int, regulationInnings: Int = 9): ProjectionResult =
        ProjectionResult(
            label = "Estimate",
            projectedFinalScore = ScoringRules.projectedFinalScore(currentScore, inningsPlayed, regulationInnings)
        )
}

File: ios/RBDarts/Features/Stats/ProjectionService.swift
import Foundation

public struct ProjectionResult: Equatable, Sendable {
    public let label: String
    public let projectedFinalScore: Double
}

public struct ProjectionService: Sendable {
    public init() {}

    public func baseline(currentScore: Int, inningsPlayed: Int, regulationInnings: Int = 9) -> ProjectionResult {
        ProjectionResult(
            label: "Estimate",
            projectedFinalScore: ScoringRules.projectedFinalScore(
                currentScore: currentScore,
                inningsPlayed: inningsPlayed,
                regulationInnings: regulationInnings
            )
        )
    }
}
